src/headers.rs

```diff
@@ -0,0 +1,179 @@
+//! This module provides access to the HTTP MIME headers
+//! of a request or response.
+//!
+//! Not to be confused with the request/response metadata
+//! which is also called "headers" in TrafficServer
+
+use crate::bindings::*;
+use crate::helpers::checked_raw_str;
+use std::marker::PhantomData;
+
+/// An iterator over header field values
+///
+/// Skips any invalid strings
+pub struct HeaderValueIter<'a, 'b, T: 'a> {
+    field: &'b HeaderField<'a, 'b, T>,
+    index: usize,
+    length: usize,
+}
+
+impl<'a, 'b, T> Iterator for HeaderValueIter<'a, 'b, T> {
+    type Item = &'a str;
+    fn next(&mut self) -> Option<Self::Item> {
+        while self.index < self.length {
+            let res = checked_raw_str(|len| unsafe {
+                TSMimeHdrFieldValueStringGet(
+                    self.field.headers.bufp,
+                    self.field.headers.offset,
+                    self.field.raw_field,
+                    self.index as i32,
+                    len,
+                )
+            });
+            self.index += 1;
+            if let Ok(res) = res {
+                return Some(res);
+            }
+        }
+        None
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (0, Some(self.length))
```

Comment:
Careful, `size_hint` wants the _remaining_ length. Returning `self.length` is thus only correct if the iterator hasn't been used.